Source/GSThemeDrawing.m

```diff
@@ -816,6 +816,139 @@ - (void) drawStepperCell: (NSCell*)cell
     [self drawStepperDownButton: downRect];
 }
 
+// NSSwitch drawing methods
+
+- (void) drawSwitchBezel: (NSRect)frame
+                forState: (NSControlStateValue)v
+                 enabled: (BOOL)enabled
+{
+  NSBezierPath *p;
+  NSPoint point;
+  CGFloat radius;
+  NSColor *backgroundColor;
+  
+  if (enabled == NO)
+    {
+      backgroundColor = [NSColor disabledControlTextColor];
+    }
+  else if (NSControlStateValueOn != v)
+    {
+      backgroundColor = [NSColor windowBackgroundColor]; // offColor
+    }
+  else
+    {
+      backgroundColor = [NSColor selectedControlColor]; // onColor
+    }
+  
+  // make smaller than enclosing frame
+  frame = NSInsetRect(frame, 4, 4);
+  radius = frame.size.height / 2.0;
+  point = frame.origin;
+  point.x += radius;
+  point.y += radius - 0.5;
+
+  // Draw initial path to enclose the button...
+  // left half-circle
+  p = [NSBezierPath bezierPath];
+  [p appendBezierPathWithArcWithCenter: point
+				radius: radius
+			    startAngle: 90.0
+			      endAngle: 270.0];
+
+  // line to first point and right halfcircle
+  point.x += frame.size.width - frame.size.height;
+  [p appendBezierPathWithArcWithCenter: point
+				radius: radius
+			    startAngle: 270.0
+			      endAngle: 90.0];
+  [p closePath];
+
+  // fill with background color
+  [backgroundColor set];
+  [p fill];
+
+  // and stroke rounded button
+  [[NSColor shadowColor] set];
+  [p stroke];
+
+  // Add highlights...
+  point = frame.origin;
+  point.x += radius - 0.5;
+  point.y += radius - 0.5;
+  p = [NSBezierPath bezierPath];
+  [p setLineWidth: 1.0];
+  [p appendBezierPathWithArcWithCenter: point
+				radius: radius
+			    startAngle: 135.0
+			      endAngle: 270.0];
+
+  // line to first point and right halfcircle
+  point.x += frame.size.width - frame.size.height;
+  [p appendBezierPathWithArcWithCenter: point
+				radius: radius
+			    startAngle: 270.0
+			      endAngle: 315.0];
+  [[NSColor controlLightHighlightColor] set];
+  [p stroke];
+}
+
+- (void) drawSwitchKnob: (NSRect)frame
+               forState: (NSControlStateValue)value
+                enabled: (BOOL)enabled
+{
+  NSColor *backgroundColor = enabled ? [NSColor windowBackgroundColor] : [NSColor disabledControlTextColor];
+  NSBezierPath *oval;
+  NSRect rect = NSZeroRect;
+  CGFloat w = (frame.size.width / 2) - 2;
+  CGFloat h = frame.size.height - 5;
```

Comment:
ON my computer the knob still has a few pixels outside of the bezel. With a value of 6 here it is completely inside. Of course this could also be achieved with other changes.

Reply:
Changed... looks excellent.

Reply:
Yes, but where is this change?

Reply:
Committed.